### PR TITLE
DOC: Improve docstring of numpy.testing.assert_allclose

### PR DESCRIPTION
**Pull request by sachu92**
DOC: Elaborate the documentation of `numpy.testing.assert_allclose` as suggested in #21578.

### CODE REVIEW

File: numpy/testing/_private/utils.py [hunk 1]
```diff
@@ -1508,6 +1515,12 @@ def assert_allclose(actual, desired, rtol=1e-7, atol=0, equal_nan=True,
     --------
     assert_array_almost_equal_nulp, assert_array_max_ulp
 
+    Notes
+    -----
+    When one of `actual` and `desired` is a scalar and the other is array_like,
```
Comment (bsipocz) [inline numpy/testing/_private/utils.py:1520]: Use double backtick around parameter names to ensure sphinx won't try to make these links.
```suggestion
    When one of ``actual`` and ``desired`` is a scalar and the other is array_like,
```

Comment (sachu92) [reply numpy/testing/_private/utils.py]: Thanks! I added the double backticks and adjusted the line lengths to within limits.

Comment (bsipocz) [reply numpy/testing/_private/utils.py]: Thanks @charris! I was only thinking about the warnings in nitpick sphinx mode and didn't know about this part of the standard.

**Comment by sachu92** [comment]
Thank you @charris for the merge!

**Pull request merged**